tests/client_tests.ts

```diff
@@ -0,0 +1,393 @@
+            uidAndPrimaryKey.primaryKey
+          )
+          expect(response).toHaveProperty('createdAt', expect.any(String))
+          expect(response).toHaveProperty('updatedAt', expect.any(String))
+        })
+    })
+    test(`${permission} key: Get all indexes when not empty`, async () => {
+      await client.listIndexes().then((response: Types.IndexResponse[]) => {
+        const indexes = response.map((index) => index.uid)
+        expect(indexes).toEqual(expect.arrayContaining([uidAndPrimaryKey.uid]))
+        expect(indexes).toEqual(expect.arrayContaining([uidNoPrimaryKey.uid]))
+      })
+    })
+    test(`${permission} key: show index with primary key`, async () => {
+      const index = client.getIndex(uidAndPrimaryKey.uid)
+      await index.show().then((response: Types.IndexResponse) => {
+        expect(response).toHaveProperty('uid', uidAndPrimaryKey.uid)
+        expect(response).toHaveProperty(
+          'primaryKey',
+          uidAndPrimaryKey.primaryKey
+        )
+      })
+    })
+
+    test(`${permission} key: show index with NO primary key`, async () => {
+      const index = client.getIndex(uidNoPrimaryKey.uid)
+      await index.show().then((response: Types.IndexResponse) => {
+        expect(response).toHaveProperty('uid', uidNoPrimaryKey.uid)
+        expect(response).toHaveProperty('primaryKey', null)
+      })
+    })
+
+    test(`${permission} key: update primary key index on an index that has no primary key already`, async () => {
+      const index = client.getIndex(uidNoPrimaryKey.uid)
+      await index
+        .updateIndex({ primaryKey: 'newPrimaryKey' })
+        .then((response: Types.IndexResponse) => {
+          expect(response).toHaveProperty('uid', uidNoPrimaryKey.uid)
+          expect(response).toHaveProperty('primaryKey', 'newPrimaryKey')
+        })
+    })
+
+    test(`${permission} key: update primary key index on an index that has already a primary key and fail`, async () => {
+      const index = client.getIndex(uidAndPrimaryKey.uid)
+      await expect(
+        index.updateIndex({ primaryKey: 'newPrimaryKey' })
+      ).rejects.toThrowError(`The primary key cannot be updated`)
+    })
+
+    test(`${permission} key: delete index`, async () => {
+      const index = client.getIndex(uidNoPrimaryKey.uid)
+      await index.deleteIndex().then((response: string) => {
+        expect(response).toBe('')
+      })
+      await expect(client.listIndexes()).resolves.toHaveLength(1)
+    })
+
+    test(`${permission} key: show deleted index should fail`, async () => {
+      const index = client.getIndex(uidNoPrimaryKey.uid)
+      await expect(index.show()).rejects.toThrowError(
+        `Index ${uidNoPrimaryKey.uid} not found`
+      )
+    })
+
+    test(`${permission} key: create index with already existing uid should fail`, async () => {
+      await expect(client.createIndex(uidAndPrimaryKey)).rejects.toThrowError(
+        `Impossible to create index; index already exists`
```

Comment:
Is it a way to check the message of the Exception? Can we check the type of the Exception too? To check that we use our own exception. But if it's too complicated, don't do it.